client/src/node.rs

```diff
@@ -0,0 +1,49 @@
+use crate::config::ClientNodeConfig;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+use warp::Filter;
+
+lazy_static! {
+    static ref DEFAULT_HEALTH_RESPONSE: HealthResponse = HealthResponse { is_healthy: true };
+}
+
+pub(crate) const DEFAULT_HEALTH_ENDPOINT: &str = "health";
+
+pub struct IPCClientNode {
+    config: ClientNodeConfig,
+}
+
+impl IPCClientNode {
+    pub fn new(config: ClientNodeConfig) -> Self {
+        Self { config }
+    }
+
+    /// Runs the node in the current thread
+    pub async fn run(&self) {
+        let json_rpc = warp::post()
+            .and(warp::path("json_rpc"))
+            .and(warp::body::bytes())
+            .map(|bytes: bytes::Bytes| {
+                log::info!("received bytes = {:?}", bytes);
+                warp::reply::json(&())
+            });
+
+        let health = warp::get()
+            .and(warp::path(DEFAULT_HEALTH_ENDPOINT))
```

Comment:
I was thinking that the same trait you use for the `cli` could be use to register the handler for the `json_rpc` server handler and the response and requests

Reply:
Oh, that might need to be more careful. Those 2 ought to be 2 traits? For cli, it is quite natural to have no states in the handler, but for `json_rpc`, the handler not only might have states but also have dependencies among each other. Also their parameters might be different as well. For example, `healthcheck` cli handler is just taking the url of which node to ping, but for `json_rpc`, it might be taking some other dependent handler or services. Might be difficult to share the same handle for both `cli` and `json_rpc`.

Reply:
Sure, sure. The idea is to have a `CliHandler` and a `JsonRPCHandler`. Actually, you can even make them independent traits for each and use a type bound in the macro (or enum) for both, so they are loaded to the CLI and spawned in the server.